### PR TITLE
Fix Using unsupported buffer type: 253 (parameter: 1) when trying to use the CLI

### PR DESCRIPTION
**Pull request by xiangzhai**
Hi @weiznich 

`brew install mysql-client` for macOS will install mysql-client 8.3, but `diesel` CLI thrown such error to Apple fans: https://github.com/diesel-rs/diesel/issues/3929

```
Using unsupported buffer type: 253 (parameter: 1)
```

I autogen [mysqlclient-sys](https://github.com/xiangzhai/mysqlclient-sys) by [rust-bindgen](https://rust-lang.github.io/rust-bindgen/) to support new and old versions about MySQL. And adjusted `diesel` to meet the `diff`.

Could you review the patch please?

Thanks,
Leslie Zhai

### CODE REVIEW

File: diesel/src/mysql/connection/bind.rs [hunk 1]
```diff
@@ -154,6 +154,11 @@ impl From<u32> for Flags {
     }
 }
 
+#[cfg(target_os = "linux")]
```
Comment (xiangzhai) [inline diesel/src/mysql/connection/bind.rs:157]: How to fix the ugly `FALSE`?

File: diesel/src/mysql/connection/bind.rs [hunk 2]
```diff
@@ -154,6 +154,11 @@ impl From<u32> for Flags {
...
+static FALSE: ffi::my_bool = 0;
+#[cfg(any(target_os = "macos", target_os = "windows"))]
```
Comment (xiangzhai) [inline diesel/src/mysql/connection/bind.rs:159]: If [Conditional compilation](https://doc.rust-lang.org/reference/conditional-compilation.html#conditional-compilation) supported `mysql_client_version` is better.

File: diesel/src/mysql/connection/bind.rs (hunk 1 above)
Comment (Mingun) [reply diesel/src/mysql/connection/bind.rs:157]: Maybe use `ffi::my_bool::default()`?

File: diesel/Cargo.toml [hunk 3]
```diff
@@ -17,7 +17,7 @@ byteorder = { version = "1.0", optional = true }
 chrono = { version = "0.4.20", optional = true, default-features = false, features = ["clock", "std"] }
 libc = { version = "0.2.0", optional = true }
 libsqlite3-sys = { version = ">=0.17.2, <0.29.0", optional = true, features = ["bundled_bindings"] }
-mysqlclient-sys = { version = "0.2.5", optional = true }
+mysqlclient-sys = { git = "https://github.com/xiangzhai/mysqlclient-sys.git", branch = "rust-bindgen", optional = true }
```
Comment (weiznich) [inline diesel/Cargo.toml:20]: It's not possible to depend on a forked git version here. These patches need to be included into the mysqlclient-sys repo.

(Open a PR there and we can discuss the changes there)

File: diesel/src/mysql/connection/bind.rs (hunk 2 above)
Comment (xiangzhai) [reply diesel/src/mysql/connection/bind.rs]: I will try [cargo::rustc-cfg=KEY[="VALUE"]](https://doc.rust-lang.org/cargo/reference/build-scripts.html#rustc-cfg)

File: diesel/Cargo.toml (hunk 3 above)
Comment (xiangzhai) [reply diesel/Cargo.toml:20]: RFR:  https://github.com/sgrif/mysqlclient-sys/pull/37

**Comment by WumaCoder** [comment]
Is there any other solution?

**Comment by weiznich** [comment]
@WumaCoder Such questions are highly discouraged at WIP PR's. If you want to contribute a fix feel free to open a new PR.

**Pull request closed without merging**